transcoder/Transcoder.py

```diff
@@ -0,0 +1,264 @@
+                    logging.info('Sampled messages: %s', self.sampling_count)
+
+                if self.message_parser.message_type_inclusions is not None:
+                    logging.info('Message type inclusions: %s', self.message_parser.message_type_inclusions)
+                elif self.message_parser.message_type_exclusions is not None:
+                    logging.info('Message type exclusions: %s', self.message_parser.message_type_exclusions)
+
+                if self.create_schemas_only is False:
+                    logging.info('Source record count: %s', self.source.record_count)
+                    logging.info('Processed record count: %s', self.message_parser.record_count)
+                    logging.info('Processed schema count: %s', self.message_parser.total_schema_count)
+                    logging.info('Summary of message counts: %s', self.message_parser.record_type_count)
+                    logging.info('Summary of error message counts: %s', self.message_parser.error_record_type_count)
+                    logging.info('Message rate: %s per second', round(self.source.record_count / total_seconds, 6))
+
+            else:
+                logging.info('Source record count: %s', self.source.record_count)
+
+            logging.info('Total runtime in seconds: %s', round(total_seconds, 6))
+            logging.info('Total runtime in minutes: %s', round(total_seconds / 60, 6))
+
+    def process_schemas(self):
+        """Process the schema specified at runtime"""
+        spec_schemas = self.message_parser.process_schema()
+        for schema in spec_schemas:
+            if len(schema.fields) == 0:
+                logging.info('Schema "%s" contains no field definitions, skipping schema creation', schema.name)
+                continue
+
+            for handler in self.all_handlers:
+                if handler.supports_all_message_types is True \
+                        or schema.message_id in handler.supported_message_types:
+                    handler.append_manufactured_fields(schema)
+
+            self.output_manager.enqueue_schema(schema)
+
+        # Only need to wait if lazy create is off, and you want to force creation before data is read
+        if self.lazy_create_resources is False:
+            self.output_manager.wait_for_schema_creation()
+
+    def handle_exception(self, raw_record, message, exception):
+        """Process exceptions encountered in the message processing runtime"""
+        if message is not None:
+            self.message_parser.increment_error_summary_count(message.name)
+        else:
+            self.message_parser.increment_error_summary_count()
+
+        self.error_writer.write_error(raw_record, message, exception)
+
+        if self.continue_on_error is False:
+            raise exception
+
+
+    def trap(self, _signum, _frame):
+        """Trap SIGINT to suppress noisy stack traces and show interim summary"""
+        print()
+        self.print_summary()
+        sys.exit(1)
```

Comment:
In the future we could make something more meaningful of the exit codes based on the summary